CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSV.java

```diff
@@ -241,25 +256,49 @@ public void setupCSV(ExtensionWebSocketClient oClient, String fileFolderPath, St
             this.config = config;
             this.options = options;
             this.oClient = oClient;
-
-            serviceWatcher = FileSystems.getDefault().newWatchService();
-            Path path = Paths.get(fileFolderPath);
-            path.register(serviceWatcher, ENTRY_CREATE);
-            log.info("CSV trying to subscribe to {}", fileFolderPath);
+            this.fileFolderPath = fileFolderPath;
 
             prepareConfigurationData();
 
-            new Thread(() -> processThread(fileFolderPath)).start();
-
-            // working on files already that exist in folder.
-            if (options.get("processExistingFiles") != null) {
-                Object processExistingFiles = options.get("processExistingFiles");
-                if (processExistingFiles instanceof Boolean)
-                    if ((boolean) processExistingFiles) {
-                        log.info("Start working on existing file in folder {}", fileFolderPath);
-                        handleExistingFiles(fileFolderPath);
-                    }
+            if (isRunnigInDocker) {
+                this.fullFilePath = fixFileFolderPathForDocker(this.fullFilePath);
+                this.fileFolderPath = fixFileFolderPathForDocker(this.fileFolderPath);
             }
+            String tmpFileFolderPath = this.fileFolderPath;
+
+            log.info("CSV Running in Docker , trying to subscribe to {} PollTime {}", this.fileFolderPath, pollTime);
+
+            TimerTask task = new TimerTask() {
+                @Override
+                public void run() {
+                    log.info("TimerTask Start working on existing file in folder {}", tmpFileFolderPath);
+                    handleExistingFiles(tmpFileFolderPath);
+
+                }
+            };
+            // Create new Timer, and schedule the task according to the pollTime
+            inDockerTime = new Timer("executePolling");
```

Comment:
This doesn't really seem to have anything to do with Docker at all. This variable should just be `timerTask` or something, shouldn't it?

Reply:
yes, a leftover evaluation of the learning curve, I will fix it. done